### PR TITLE
Code journal project features 1. Adding entries. 

### PR DESCRIPTION
**Pull request by KennethDelaCruz**
code journal feature 1

### CODE REVIEW

File: css/styles.css [hunk 1]
```diff
@@ -0,0 +1,44 @@
+img {
+  max-width: 100%;
+  max-height: 100%;
```
Comment (codingcodymiller) [inline css/styles.css:7]: Double check if this is doing anything.

File: index.html [hunk 2]
```diff
@@ -16,8 +16,32 @@
...
+    <div class="container" data-view="entry-form">
+      <form>
+        <h2>New Entry</h2><br />
+        <div class="row">
+          <div class="column-half">
+          <img src="images/placeholder-image-square.jpg">
+          </div>
+        <div class="column-half">
+          <label for="title">Title:</label><br />
+          <input type="text" id="new-post-title" name="title" required><br />
```
Comment (codingcodymiller) [inline index.html]: Change the `id` here. It should describe that this is the `input` for the title, not the title itself.

File: index.html [hunk 3]
```diff
@@ -16,8 +16,32 @@
...
+          <label for="img">Image URL:</label><br />
+          <input type="text" id="new-post-img" name="img" required><br />
```
Comment (codingcodymiller) [inline index.html]: Change the `id` here. It should describe that this is the `input` for the image, not the image itself.

File: index.html [hunk 4]
```diff
@@ -16,8 +16,32 @@
...
   <main>
+    <div class="container" data-view="entry-form">
+      <form>
+        <h2>New Entry</h2><br />
+        <div class="row">
+          <div class="column-half">
+          <img src="images/placeholder-image-square.jpg">
+          </div>
+        <div class="column-half">
+          <label for="title">Title:</label><br />
```
Comment (codingcodymiller) [inline index.html]: Your `for` attribute values don't match the `id` for their form control. You can always test this by clicking on  a label. If it is set up correctly, it should `focus` the input it is tied to.

File: index.html [hunk 5]
```diff
@@ -16,8 +16,32 @@
...
   <header>
+    <h2 class="container">Code Journal</h2>
   </header>
   <main>
+    <div class="container" data-view="entry-form">
+      <form>
+        <h2>New Entry</h2><br />
+        <div class="row">
+          <div class="column-half">
+          <img src="images/placeholder-image-square.jpg">
```
Comment (codingcodymiller) [inline index.html]: Indent here

File: index.html [hunk 6]
```diff
@@ -16,8 +16,32 @@
...
+          </div>
+        <div class="column-half">
+          <label for="title">Title:</label><br />
+          <input type="text" id="new-post-title" name="title" required><br />
+          <label for="img">Image URL:</label><br />
+          <input type="text" id="new-post-img" name="img" required><br />
+        </div>
+        </div><br />
```
Comment (codingcodymiller) [inline index.html]: This `br` element seems unnecessary. If it is for spacing, use margin.

File: js/data.js [hunk 7]
```diff
@@ -6,3 +6,8 @@ var data = {
   editing: null,
   nextEntryId: 1
 };
+
+var previousData = localStorage.getItem('data model');
```
Comment (codingcodymiller) [inline js/data.js]: Don't have spaces in your localStorage property name. Use a `-` to delimit your words

File: js/main.js [hunk 8]
```diff
@@ -1,2 +1,39 @@
...
+
+imgInput.addEventListener('input', imgPreview);
+formRoute.addEventListener('submit', submitForm);
+
+function saveLocal(event) {
+  var dataJSON = JSON.stringify(data);
+  localStorage.setItem('data model', dataJSON);
+}
+
+window.addEventListener('beforeunload', saveLocal);
```
Comment (codingcodymiller) [inline js/main.js]: This should be in `data.js`

**Review by codingcodymiller — CHANGES_REQUESTED** [review]
Great work so far! Please see my notes on your code. Also, here are a few notes here on things to fix:
1. Add some vertical whitespace to the header
2. Add some height to your inputs
3. Right-align the "Submit" button
4. Change your font family to something other than Times New Roman 🤮

**Review by codingcodymiller — CHANGES_REQUESTED** [review]
Hey Kenn! Please attach an updated screenshot of your output after the changes I requested

**Comment by KennethDelaCruz** [comment]
code journal feature 1

**Comment by KennethDelaCruz** [comment]
Here is the GIF with the  updates!
code journal feature 1

**Review by codingcodymiller — APPROVED** [review]
🎉 🎉 🎉

**Pull request merged**